src/coreclr/vm/ceeload.cpp

```diff
@@ -4722,8 +4722,59 @@ VASigCookie *Module::GetVASigCookie(Signature vaSignature)
         {
             if (pBlock->m_cookies[i].signature.GetRawSig() == vaSignature.GetRawSig())
             {
-                pCookie = &(pBlock->m_cookies[i]);
-                break;
+                if (pBlock->m_cookies[i].IsUnloaded())
+                {
+                    continue;
+                }
+
+                const SigTypeContext& leftTypeContext = pBlock->m_cookies[i].typeContext;
+                _ASSERTE(leftTypeContext.m_classInst.GetNumArgs() == typeContext->m_classInst.GetNumArgs());
+                _ASSERTE(leftTypeContext.m_methodInst.GetNumArgs() == typeContext->m_methodInst.GetNumArgs());
+
+                bool instMatch = true;
+
+                for (UINT j = 0; j < leftTypeContext.m_classInst.GetNumArgs(); j++)
+                {
+                    if (leftTypeContext.m_classInst[j].GetAssembly()->IsCollectible()
+                        && leftTypeContext.m_classInst[j].GetLoaderAllocator()->IsUnloaded())
```

Comment:
`m_classInst` can be physically unloaded from memory at this point. Calling methods on it can crash with access violation.

The way to handle unloadability is by allocating the cookie on the loader allocator with the right lifetime. I think you will want to move this hashtable to loader allocator.